content/en/blog/_posts/2025-04-17-volume-populators-ga/index.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: blog
+title: "Kubernetes 1.33: Volume Populators Graduate to GA"
+date: 2025-04-17
+slug: volume-populators-ga
+author: >
+  Danna Wang (Google)
+  Sunny Song (Google)
+---
+
+The volume populators feature is now entering GA! The `AnyVolumeDataSource` feature
+gate defaults to enabled in Kubernetes v1.33, which means that users can specify any custom resource
+as the data source of a PVC.
+
+## What is new
+
+There are three major enhancements from beta.
+
+### Populator Pod is optional
+
+During the beta phase, we discovered that initiating data population doesn't always require a dedicated populator pod. In some cases, it's more efficient to interact with the cloud provider's APIs directly from the controller pod.
+
+To accommodate this, we've introduced three new plugin-based functions:
+* PopulateFn(): Executes the provider-specific data population logic.
+* PopulateCompleteFn(): Checks if the data population operation has finished successfully.
+* PopulateCleanupFn(): Cleans up temporary resources created by the provider-specific functions after data population is completed
+
+### Mutator functions to modify the Kubernetes resources
+
+For GA a MutatorConfig to the Volume Populator Controller is added, allowing the specification of mutator functions to modify Kubernetes resources. This enables customization of the default Kubernetes objects in the volume populator.
```

Comment:
Can you elaborate why mutation is needed?  What's the use case?

Reply:
The PVC object only has:
```
dataSourceRef:
    apiGroup: provider.example.com
    kind: Provider
    name: provider1
```
as extra. But in the actual volume populator you can set provisioner as part of MutatorConfig.

Reply:
Sure.  Add this to the blog to clarify?

Reply:
added to the diff here: https://github.com/kubernetes/website/pull/49542/commits/936e2a33ae11aea28ed267b2122f9391e78a1061